src/Pages/Templates/TemplateDetails/components/Tabs/TemplateSystemsTab.tsx

```diff
@@ -0,0 +1,311 @@
+import { useEffect, useMemo, useState } from 'react';
+import { createUseStyles } from 'react-jss';
+import { Outlet, useNavigate, useParams } from 'react-router-dom';
+import { useQueryClient } from 'react-query';
+import { SortByDirection, type ThProps } from '@patternfly/react-table';
+import Loader from 'components/Loader';
+import {
+  useDeleteTemplateFromSystems,
+  useListSystemsByTemplateId,
+} from 'services/Systems/SystemsQueries';
+import SystemsTable from 'components/SharedTables/SystemsTable';
+import EmptyTableState from 'components/EmptyTableState/EmptyTableState';
+import ConditionalTooltip from 'components/ConditionalTooltip/ConditionalTooltip';
+import { useAppContext } from 'middleware/AppContext';
+import { ADD_ROUTE } from 'Routes/constants';
+import Hide from 'components/Hide/Hide';
+import SystemsDeleteKebab from 'components/SharedTables/SystemsTable/Components/SystemsDeleteKebab';
+import { SearchIcon } from '@patternfly/react-icons';
+import useDebounce from 'Hooks/useDebounce';
+
+const useStyles = createUseStyles({
+  description: {
+    paddingTop: '12px', // 4px on the title bottom padding makes this the "standard" 16 total padding
+    paddingBottom: '8px',
+    color: global_Color_200.value,
+  },
+  mainContainer: {
+    backgroundColor: global_BackgroundColor_100.value,
+    display: 'flex',
+    flexDirection: 'column',
+  },
+  topContainer: {
+    justifyContent: 'space-between',
+    padding: '16px 24px',
+    height: 'fit-content',
+    flexDirection: 'row!important',
+    minHeight: '68px', // Prevents compacting of the search box (patternfly bug?)
+  },
+  topContainerWithFilterHeight: { extend: 'topContainer', minHeight: '128px' },
+  bottomContainer: {
+    justifyContent: 'space-between',
+    minHeight: '68px',
+  },
+  // Needed to fix styling when "Add repositories" button is disabled
+  ctions: {
+    display: 'flex',
+    flexDirection: 'row',
+  },
+});
+
+const perPageKey = 'TemplateSystemsPerPage';
+
+export default function TemplateSystemsTab() {
+  const classes = useStyles();
+  const queryClient = useQueryClient();
+  const navigate = useNavigate();
+  const { templateUUID: uuid } = useParams();
+  const { rbac } = useAppContext();
+
+  const storedPerPage = Number(localStorage.getItem(perPageKey)) || 20;
+  const [page, setPage] = useState(1);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [perPage, setPerPage] = useState(storedPerPage);
+
+  const [activeSortIndex, setActiveSortIndex] = useState<number>(-1);
+  const [activeSortDirection, setActiveSortDirection] = useState<'asc' | 'desc'>('asc');
+  const [selected, setSelected] = useState<string[]>([]);
+  const selectedList = useMemo(() => new Set(selected), [selected]);
+
+  const debouncedSearchQuery = useDebounce(searchQuery, searchQuery ? 400 : 0);
+
+  useEffect(() => {
+    setPage(1);
+  }, [debouncedSearchQuery]);
+
+  const columnSortAttributes = [
```

Comment:
was sorting disabled for some columns intentionally?

Reply:
Yes. Some of the sorting doesn't seem to work as expected (groups/workspace seems random, the advisories columns are added together, etc.)